File: wrappers/pypi/tests/fake_skillxp.py
#!/usr/bin/env python3
"""Fake skillxp binary for wrapper tests: behavior is driven by FAKE_*
env vars so tests can point SKILLXP_BINARY at a controllable stand-in."""

import os
import sys

sys.stdout.write(os.environ.get("FAKE_STDOUT", ""))
sys.stderr.write(os.environ.get("FAKE_STDERR", ""))
sys.exit(int(os.environ.get("FAKE_EXIT", "0")))

File: wrappers/npm/test/fake-skillxp.js
#!/usr/bin/env node
"use strict";

// Fake skillxp binary for wrapper tests: behavior is driven by FAKE_* env
// vars so tests can assert the shim forwards stdio and exit codes
// untouched.
process.stdout.write(process.env.FAKE_STDOUT || "");
process.stderr.write(process.env.FAKE_STDERR || "");
process.exit(Number(process.env.FAKE_EXIT || 0));
